health_vara.py

```diff
@@ -189,6 +239,20 @@ def set_passthrough_field(cls, patients, patient_field_name, value):
         party_field_name = get_party_field_name(patient_field_name)
         Party.write([patient.name for patient in patients], {party_field_name: value})
 
+    # Overrides to add in the general search by 'partner_patient_id', otherwise identical
+    @classmethod
+    def search_rec_name(cls, _name, clause):
+        if clause[1].startswith('!') or clause[1].startswith('not '):
+            bool_op = 'AND'
+        else:
+            bool_op = 'OR'
+        return [bool_op,
+                ('puid',) + tuple(clause[1:]),
+                ('name',) + tuple(clause[1:]),
+                ('lastname',) + tuple(clause[1:]),
+                ('partner_patient_id',) + tuple(clause[1:]),
```

Comment:
this is copy-pasted from gnuhealth code, except for this one clause, which allows us to also look for the value as a `partner_patient_id`.

Reply:
I saw there is a new `search_mobile` function defined for making the phone number searchable. Does it also need to be included here along with the `partner_patient_id` to become usable in the general search?

Reply:
Thanks for explaining this @DanielStephens 👍 In case somebody else stumbles over it: the phone number is searchable without an explicit mention here, because `name` references the entire party object, which contains the phone number and therefore makes it transitively searchable.